manifests/config.pp

```diff
@@ -57,6 +56,13 @@
     }
   }
 
+  file { $_server_conf_dir:
+    ensure => 'directory',
+    owner  => $keycloak::user,
+    group  => $keycloak::group,
+    mode   => '0750',
+  }
```

Comment:
I'd put this file back where it was, the order in manifest doesn't matter, though might by chance make things work, what is really needed is `require` line like one of your commits had.  Puppet does not guarantee that order in manifests is order of execution so need to set hard dependencies.

Reply:
I think if you revert `dc63f7a08fa4b32750c13704455baaf83ddb715f` should be good.